File: Java/src/main/java/Solution001_100/Solution70.java
package Solution001_100;

/**
 * @author windows
 */
public class Solution70 {
    public int climbStairs(int n) {
        int[] df = new int[n];
        df[0] = 1;
        df[1] = 1;
        for (int i = 2; i < n; i++) {
            df[i] = df[i - 1] + df[i - 2];
        }
        return df[n - 1];
    }


    public static void main(String[] args) {
        Solution70 solution70 = new Solution70();
        System.out.println(solution70.climbStairs(4));
    }
}

File: C++/src/Solution1_100/Solution70.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution70.h"
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

int climbStairs(int n) {
    int dp[n + 1];
    dp[1] = 1;
    dp[0] = 1;
    for (int i = 2; i <= n; ++i) {
        dp[i] = dp[i - 1] + dp[i - 2];
    }
    return dp[n];
}
